tests/Unit/Libraries/Cms/Schema/MysqlChangeItemTest.php

```diff
@@ -0,0 +1,511 @@
+<?php
+/**
+ * @package     Joomla.UnitTest
+ * @subpackage  Schema
+ *
+ * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Tests\Unit\Libraries\Cms\Schema;
+
+use Joomla\CMS\Schema\ChangeItem\MysqlChangeItem;
+use Joomla\Database\Mysqli\MysqliDriver;
+use Joomla\Tests\Unit\UnitTestCase;
+
+/**
+ * Test class for \Joomla\CMS\Schema\ChangeItem\MysqlChangeItem
+ *
+ * @package     Joomla.UnitTest
+ * @subpackage  Schema
+ *
+ * @testdox     The MysqlChangeItem
+ *
+ * @since       __DEPLOY_VERSION__
+ */
+class MysqlChangeItemTest extends UnitTestCase
+{
+	/**
+	 * @testdox  can build the right query
+	 *
+	 * @dataProvider  constructData
+	 *
+	 * @param   array  $options  Options array to inject
+	 * @param   array  $expects  Expected data values
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function testBuildCheckQuery($options, $expects)
+	{
+		$message = "Test '%s' for query '". $options['query'] . "'";
+		$utf8mb4 = $options['utf8mb4'] ?? true;
+
+		if (!$utf8mb4)
+		{
+			$message .= ' without utf8mb4 support';
+		}
+
+		$message .= ' failed.';
+
+		$db = $this->createStub(MysqliDriver::class);
+		$db->method('getServerType')->willReturn('mysql');
+		$db->method('getPrefix')->willReturn('jos_');
+		$db->method('hasUTF8mb4Support')->willReturn($utf8mb4); 
+		$db->method('quote')->will(
+			$this->returnCallback(function ($arg) {
+				return "'" . $arg . "'";
+			})
+		);
+		$db->method('quoteName')->will(
+			$this->returnCallback(function ($arg) {
+				return '`' . $arg . '`';
+			})
+		);
+
+		$item = new MysqlChangeItem($db, '', $options['query']);
+
+		$this->assertEquals($expects['checkQuery'], $item->checkQuery, sprintf($message, 'checkQuery'));
+		$this->assertEquals($expects['queryType'], $item->queryType, sprintf($message, 'queryType'));
+		$this->assertEquals($expects['checkQueryExpected'], $item->checkQueryExpected, sprintf($message, 'checkQueryExpected'));
+		$this->assertEquals($expects['msgElements'], $item->msgElements, sprintf($message, 'msgElements'));
+		$this->assertEquals($expects['checkStatus'], $item->checkStatus, sprintf($message, 'checkStatus'));
+	}
+
+	/**
+	 * Provides constructor data for test methods
+	 *
+	 * @return  array
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function constructData(): array
```

Comment:
From a maintainer perspective I would turn this data into functions. So everybody else will understand whyt is actually tested. Like that it is hard to understand. I know it is a bit more work, but for others easier to follow.

Reply:
@laoneo I don't understand what you mean with "turn this data into functions". Do you have an example where I can see that for a similar use case?

Reply:
@laoneo Or do you simply mean I shall not use that data array and data provider method at all and make a test method for each query?

Reply:
Then it would be much more understandable. But not sure if you want to do that effort.

Reply:
It could be some 100 methods in case of MySQL at the end, with all the overhead code lines for doc blocks and so on. But I understand what you mean.

Reply:
I could also do a mix of both, separate test methods for the different kinds of statements, like create table, alter table add column, alter table modify column and so on, and then do the different variants of these with data providers. Would this be a good idea? Or silly? In any case with different test methods we will have lots of duplicated code, what I wanted to avoid with the data provider and data array way.

Reply:
I will prepare something. It can take a while. Maybe I will do it in another branch and make an alternative dummy PR.